tests/FileStorageTest.php

```diff
@@ -14,25 +14,264 @@
+        }
+
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->client->rotateAbsFileStorageCredentials($storage['id'], self::ROTATE_ABS_OPTIONS);
+    }
+
+    public function testListS3Storages()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $initCount = count($this->client->listS3FileStorage());
+        $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+        $storages = $this->client->listS3FileStorage();
+
+        $this->assertSame($initCount + 1, count($storages));
+
+        foreach ($storages as $storage) {
+            if ($storage['provider'] === 'azure') {
+               $this->fail('List of S3 storages contains also Azure Blob Storages');
+            }
+        }
+    }
+
+    public function testListAbsStorages()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $initCount = count($this->client->listAbsFileStorage());
+        $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+        $storages = $this->client->listAbsFileStorage();
+
+        $this->assertSame($initCount + 1, count($storages));
+
+        foreach ($storages as $storage) {
+            if ($storage['provider'] === 'aws') {
+               $this->fail('List of Azure Blob Storages contains also S3 Storage');
+            }
+        }
+    }
+
+    public function testSetS3StorageAsDefault()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $this->assertFalse($storage['isDefault']);
+
+        $storage = $this->client->setS3FileStorageAsDefault($storage['id']);
+
+        $this->assertTrue($storage['isDefault']);
+    }
+
+    public function testSetAbsStorageAsDefault()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->assertFalse($storage['isDefault']);
+
+        $storage = $this->client->setAbsFileStorageAsDefault($storage['id']);
+
+        $this->assertTrue($storage['isDefault']);
     }
 
-    public function testProjectAssignFileStorage()
+    public function testCrossProviderStorageDefaultAbsS3()
     {
-        $this->markTestSkipped('just for development');
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('File storage with id "%d" and provider "aws" was not found.', $storage['id']));
```

Comment:
oddělit od sebe neexistující ID a to, že ID je jiného typu, než  se očekává

Reply:
i u azure

Reply:
a u rotate

Reply:
Já bych tohle nerozlišoval. Tím, že je to na různým endpointu se jedná o "jiný entity". Je to jako šahat na "/getSnowflakeBeckend/1" ... -> "Backend s id 1 je RS, šáhni si pro něj na jiný endpoint".

Reply:
`AWS S3 file storage "%d" not found.`?

Reply:
@ondrejhlavacek  jo ta hláška se mi líbí víc.